packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -0,0 +1,369 @@
+  /// layer tree.
+  ///
+  /// If the associated layer is not removed, and receives a hit, then
+  /// [collectMousePositions] will assert the next time it is called.
+  void detachAnnotation(MouseDetectorAnnotation annotation) {
+    final _TrackedAnnotation trackedAnnotation = _findAnnotation(annotation);
+    assert(trackedAnnotation != null, "Tried to detach an annotation that wasn't attached: $annotation");
+    if (trackedAnnotation.activeDevices.isNotEmpty) {
+      for (int deviceId in trackedAnnotation.activeDevices) {
+        annotation.onExit(
+          MouseExitDetails(
+            globalPosition: _lastMouseEvent[deviceId]?.position,
+            sourceTimeStamp: _lastMouseEvent[deviceId]?.timeStamp,
+            deviceId: deviceId,
+          ),
+        );
+      }
+    }
+    _trackedAnnotations.remove(trackedAnnotation);
+  }
+
+  void _scheduleMousePositionCheck() {
+    SchedulerBinding.instance.addPostFrameCallback((Duration _) => collectMousePositions());
+    SchedulerBinding.instance.scheduleFrame();
+  }
+
+  // Handler for events coming from the PointerRouter.
+  void _handleEvent(PointerEvent event) {
+    if (event.kind != PointerDeviceKind.mouse) {
+      return;
+    }
+    final int deviceId = event.device;
+    if (_trackedAnnotations.isEmpty) {
+      // If we're not tracking anything, then there is no point in registering a
+      // frame callback or scheduling a frame. By definition there are no active
+      // annotations that need exiting, either.
+      _lastMouseEvent.remove(deviceId);
+      return;
+    }
+    if (event is PointerRemovedEvent) {
+      _lastMouseEvent.remove(deviceId);
+      // If the mouse was removed, then we need to schedule one more check to
+      // exit any annotations that were active.
+      _scheduleMousePositionCheck();
+    } else {
+      if (event is PointerMoveEvent || event is PointerHoverEvent || event is PointerDownEvent) {
+        if (!_lastMouseEvent.containsKey(deviceId) || _lastMouseEvent[deviceId].position != event.position) {
+          // Only schedule a frame if we have our first event, or if the
+          // location of the mouse has changed.
+          _scheduleMousePositionCheck();
+        }
+        _lastMouseEvent[deviceId] = event;
+      }
+    }
+  }
+
+  _TrackedAnnotation _findAnnotation(MouseDetectorAnnotation annotation) {
+    final _TrackedAnnotation trackedAnnotation = _trackedAnnotations[annotation];
+    assert(
+        trackedAnnotation != null,
+        'Unable to find annotation $annotation in tracked annotations. '
+        'Check that attachAnnotation has been called for all annotated layers.');
+    return trackedAnnotation;
+  }
+
+  /// Tells interested objects that a mouse has entered, exited, or moved, given
+  /// a callback to fetch the [MouseDetectorAnnotation] associated with a global
+  /// offset.
+  ///
+  /// This is called from the [RenderView] when the layer tree has been updated,
```

Comment:
Is this comment still accurate? Seems like its only used as a postframecallback?